gordo_components/builder/build_model.py

```diff
@@ -189,6 +218,15 @@ def build_model(
     }
 
     metadata["model"].update(_get_metadata(model))
+
+    # Log metadata to MLflow
+    mlflow_client = get_mlflow_client(azureml_config)
+    run_id = get_run_id(
+        mlflow_client, experiment_name=name, model_key=azure_config["model_key"]
+    )
+    mlflow_client.log_batch(run_id, **get_batch_kwargs(metadata))
+    mlflow_client.set_terminated(run_id)
+
```

Comment:
So that argument above, it seems like one could rather create a decorator/wrapper to this function which snaps up any input required, ie calculating the 'model_key', executing the function, grabbing the metadata, logging it, and passing it along. 

Advantage here is we're not integrating _any_ specific logging client, and sets us up for different or multiple logging clients without also in twinning them into this function.

Reply:
After looking at the decorator I setup again, it can be used it as a wrapper type thing with some slight changes, while still being able to use it as a decorator if desired. I'll just put it setup like that and see what you think.